File: TransCoder_tokenized_test_set_functions/python/CHECK_TWO_STRINGS_K_ANAGRAMS_NOT.py
def arekAnagrams ( str1, str2, k ) :
    n = len ( str1 )
    if ( len ( str2 ) != n ) :
        return False
    count1 = [ 0 ] * MAX_CHAR
    count2 = [ 0 ] * MAX_CHAR
    for i in range ( n ) :
        count1 [ ord ( str1 [ i ] ) - ord ( 'a' ) ] += 1
    for i in range ( n ) :
        count2 [ ord ( str2 [ i ] ) - ord ( 'a' ) ] += 1
    count = 0
    for i in range ( MAX_CHAR ) :
        if ( count1 [ i ] > count2 [ i ] ) :
            count = count + abs ( count1 [ i ] - count2 [ i ] )
    return ( count <= k )

File: TransCoder_tokenized_test_set_functions/cpp/CHECK_TWO_STRINGS_K_ANAGRAMS_NOT.cpp
bool arekAnagrams ( string str1, string str2, int k ) {
  int n = str1 . length ( );
  if ( str2 . length ( ) != n ) return false;
  int count1 [ MAX_CHAR ] = {
    0 };
    int count2 [ MAX_CHAR ] = {
      0 };
      for ( int i = 0;
      i < n;
      i ++ ) count1 [ str1 [ i ] - 'a' ] ++;
      for ( int i = 0;
      i < n;
      i ++ ) count2 [ str2 [ i ] - 'a' ] ++;
      int count = 0;
      for ( int i = 0;
      i < MAX_CHAR;
      i ++ ) if ( count1 [ i ] > count2 [ i ] ) count = count + abs ( count1 [ i ] - count2 [ i ] );
      return ( count <= k );
    }